### PR TITLE
Add --weights as load option

### PR DESCRIPTION
**Pull request by Beanow**
This will let you use the same format `weights.json` file as the frontend allows you to upload/download and use it doing the load command.

### CODE REVIEW

**Comment by Beanow** [comment]
Edit: realized we use `scripts/build_static_site.sh` so I've added the option there as well.

File: src/cli/load.js [hunk 1]
```diff
@@ -177,7 +191,7 @@ export type SaveGraph = (
   RepoId
 ) => Promise<Graph>;
 
-export type SaveCred = (Graph, RepoId) => Promise<void>;
+export type SaveCred = (Graph, RepoId, string | null) => Promise<void>;
```
Comment (decentralion) [inline src/cli/load.js]: To make this a little clearer to the reader, how about we make it:

```js
export type SaveCred = (graph: Graph, repoId: RepoId, weightsPath?: string) => Promise<void>
```

this way someone reading the type signature can see what the string argument means. (Another option would be to make a type alias `export type WeightsPath = string)

Using `weightsPath?` means that it's an optional arg, so we won't need to do the `| null` stuff.

File: src/cli/load.js [hunk 2]
```diff
@@ -88,6 +92,7 @@ function die(std, message) {
 export type LoadOptions = {|
   +output: RepoId,
   +repoIds: $ReadOnlyArray<RepoId>,
+  +weightsPath?: string | null,
```
Comment (decentralion) [inline src/cli/load.js]: if using `weightsPath?` then you don't need to do `string | null`, since the `?` already makes it optional.

File: src/cli/load.js [hunk 3]
```diff
@@ -220,22 +234,42 @@ export const makeLoadDefaultPlugins = (
     }
     addToRepoIdRegistry(options.output);
     const graph = await saveGraph(defaultAdapterLoaders(), options.output);
-    await saveCred(graph, options.output);
+    await saveCred(graph, options.output, options.weightsPath || null);
```
Comment (decentralion) [inline src/cli/load.js]: if we make it an optional arg (see above) then we won't need the `|| null`

File: src/cli/load.js [hunk 4]
```diff
@@ -220,22 +234,42 @@ export const makeLoadDefaultPlugins = (
...
+  }
+
+  const raw = await fs.readFile(path, "utf-8");
+  return weightsFromJSON(JSON.parse(raw));
+};
+
+export const saveCred = async (
+  graph: Graph,
+  repoId: RepoId,
+  weightsPath: string | null
```
Comment (decentralion) [inline src/cli/load.js]: as above, can be `weightsPath?: string`

File: src/cli/load.js (hunk 1 above)
Comment (Beanow) [reply src/cli/load.js]: Changed it.

File: src/cli/load.js (hunk 2 above)
Comment (Beanow) [reply src/cli/load.js]: Changed it.

File: src/cli/load.js (hunk 3 above)
Comment (Beanow) [reply src/cli/load.js]: Changed it.

File: src/cli/load.js (hunk 4 above)
Comment (Beanow) [reply src/cli/load.js]: Changed it.

File: src/cli/load.js [hunk 5]
```diff
@@ -122,6 +128,14 @@ export function makeLoadCommand(
           explicitOutput = stringToRepoId(args[i]);
           break;
         }
+        case "--weights": {
+          if (weightsPath != undefined)
```
Comment (Beanow) [inline src/cli/load.js]: In general an optional field `weightsPath?: string` doesn't match nulls so I used undefined here.

File: src/cli/load.test.js [hunk 6]
```diff
@@ -544,7 +553,27 @@ describe("cli/load", () => {
         repoIds: [fooBar, fooBaz],
       });
       expect(saveCred).toHaveBeenCalledTimes(1);
-      expect(saveCred).toHaveBeenCalledWith(graph, fooCombined);
+      expect(saveCred).toHaveBeenCalledWith(graph, fooCombined, undefined);
+    });
+
+    it("calls saveCred with the weight path if set in the options", async () => {
```
Comment (Beanow) [inline src/cli/load.test.js]: Along the lines of the previous test case, I've included the case where a path _is_ set.

**Comment by Beanow** [comment]
The tests passed for me locally, I'm not sure what's up with the CI one.

**Comment by Beanow** [comment]
Rebased from master, which fixed the CI tests.

**Comment by decentralion** [comment]
CI is having issues. You can ignore ENOMEM issues or `out of retries, exiting` issues from flow. It started happening after I upgraded all the deps, I still need to look into it.

File: src/cli/load.test.js [hunk 7]
```diff
@@ -302,7 +308,10 @@ describe("cli/load", () => {
           successCase({
             name: "with a multiple repositories",
             args: ["foo/bar", "bar/zod", "--output", "bar/zod"],
-            loadOptions: {output: barZod, repoIds: [fooBar, barZod]},
+            loadOptions: {
```
Comment (decentralion) [inline src/cli/load.test.js]: I'm confused where all these unrelated changes are coming from (prettier doesn't usually churn the codebase).

Comment (Beanow) [reply src/cli/load.test.js]: It failed on the prettier check on `yarn test`, this is the result of `yarn prettify`.

Comment (decentralion) [reply src/cli/load.test.js]: interesting; on my machine, both before and after were `pretty` enough. we must have slightly different config.

**Comment by decentralion** [comment]
@Beanow I'd like this to get merged as squash+merge (rather than rebase+merge). Would you write up a commit message for the combined commit? Something that just has some context on the change and how it was tested.

**Comment by Beanow** [comment]
Add --weights as load option

Includes a change to cli load and build_static_site.sh to accept a --weights WEIGHTS_FILE argument.
This allows overriding the default weights at build-time using a weights.json that has the same format as previously generated in the frontend.

Adds an additional test-case as well for propagating the optional parameter.
The file I/O of loading and parsing a weights.json file was tested manually. As analysis/weights' fromJSON() is tested elsewhere as is passing weight parameters.
